Azure instance module refactor update

praveenramuu opened the pull request:
— multiple disk , single disk logic update
    — Inventory creation update
    — Generalize variables
    - ASG and NSG rule creation

praveenramuu committed:
REBASE
Azure instance module refactor update
    — multiple disk , single disk logic update
    — Inventory creation update
    — Generalize variables
    - ASG and NSG rule creation

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

sasikeda commented on resources/azure/compute/vm_0_disk/vm_0_disk.tf:
```diff
@@ -49,60 +41,31 @@ data "template_cloudinit_config" "user_data64" {
   }
 }
 
-resource "azurerm_private_dns_a_record" "itself" {
```
May i know why DNS entries are removed ?

praveenramuu replied:
Added DSN , Done

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -1,11 +1,30 @@
+  block_device_names = ["/dev/sdc", "/dev/sdd", "/dev/sdf", "/dev/sdg",
+  "/dev/sdh", "/dev/sdi", "/dev/sdj", "/dev/sdk", "/dev/sdl", "/dev/sdm", "/dev/sdn", "/dev/sdo", "/dev/sdp", "/dev/sdq"]
+  gpfs_base_rpm_path = var.spectrumscale_rpms_path == null ? fileset(var.spectrumscale_rpms_path, "gpfs.base-*") : null
+  ssd_device_names   = [for i in range(var.scratch_devices_per_storage_instance) : "/dev/nvme0n${i + 1}"]
```
Is this true ? looks gcp ?

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+}
+
+# Add tcp inbound security rule
+module "scale_cluster_tcp_inbound_security_rule" {
```
Any specific reason for separate  modules for tcp and udp

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+}
+
+# Create bastion-scale cluster nsg
+module "bastion_scale_cluster_nsg" {
```
Any reason for Bastion NSG to be in the instance module

praveenramuu replied:
Thanks , removed

sasikeda replied:
This is not removeed yet

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+}
+
+# Add tcp inbound security rule to allow bastion to scale communication
+module "bastion_scale_cluster_tcp_inbound_security_rule" {
```
this needs to be in conjunction with using_jumphost_connection
and need cloud-vm asg

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+  resource_group_name                        = var.resource_group_name
+}
+
+module "storage_private_dns_zone" {
```
The DNS creation should not be made compulsory, if the user already has a zone, should be used

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+  vnet_zone_link_name   = format("%s-comp-link", var.resource_prefix)
+}
+
 module "proximity_group" {
```
proximity group is optional. only required for stratch and only applicable to storage.

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+  proximity_placement_group_id  = length(var.vpc_availability_zones) > 1 ? null : module.proximity_group.proximity_group_compute_id
+  os_disk_caching               = var.compute_cluster_os_disk_caching
+  os_storage_account_type       = var.compute_boot_disk_type
+  user_public_key               = var.create_remote_mount_cluster == true ? var.compute_cluster_ssh_public_key : var.storage_cluster_ssh_public_key
```
Just to confirm, if it is key pair name or ssh content

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+    vm_name => {
+      zone   = length(var.vpc_availability_zones) > 1 ? element(slice(var.vpc_availability_zones, 0, 2), idx) : element(var.vpc_availability_zones, idx)                                        # Consider only first 2 elements in multi-az
+      subnet = length(var.vpc_storage_cluster_private_subnets) > 1 ? element(slice(var.vpc_storage_cluster_private_subnets, 0, 2), idx) : element(var.vpc_storage_cluster_private_subnets, idx) # Consider only first 2 elements
+      disks  = toset([for i in range(1, var.block_devices_per_storage_instance + 1) : "${vm_name}-data-${i}"])                                                                                  // TODO# later use the disk                                                                              # Persistent disk names
```
This needs to change w.rto #473

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
-  data_disk_size                  = var.data_disk_size
-  data_disk_storage_account_type  = var.data_disk_storage_account_type
+  os_storage_account_type         = var.storage_boot_disk_type
+  data_disks_per_storage_instance = var.block_devices_per_storage_instance
```
I think here not using the map created above, as size and type can vary per disk

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+  write_inventory                           = (var.create_remote_mount_cluster == true && local.cluster_type == "compute" || local.cluster_type == "combined") ? 1 : 0
+  clone_complete                            = module.prepare_ansible_configuration.clone_complete
+  inventory_path                            = format("%s/compute_cluster_inventory.json", var.scale_ansible_repo_clone_path)
+  cloud_platform                            = jsonencode("GCP")
```
GCP ?

praveenramuu replied:
Done , removed

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+  write_inventory                           = (var.create_remote_mount_cluster == true && local.cluster_type == "storage" || local.cluster_type == "combined") ? 1 : 0
+  clone_complete                            = module.prepare_ansible_configuration.clone_complete
+  inventory_path                            = format("%s/storage_cluster_inventory.json", var.scale_ansible_repo_clone_path)
+  cloud_platform                            = jsonencode("GCP")
```
Same as above

praveenramuu replied:
Done removed

sasikeda commented on azure_scale_templates/sub_modules/instance_template/main.tf:
```diff
@@ -16,289 +35,535 @@ module "generate_storage_cluster_keys" {
+  write_inventory                           = (var.create_remote_mount_cluster == false && local.cluster_type == "combined") ? 1 : 0
+  clone_complete                            = module.prepare_ansible_configuration.clone_complete
+  inventory_path                            = format("%s/cluster_inventory.json", var.scale_ansible_repo_clone_path)
+  cloud_platform                            = jsonencode("GCP")
```
Same as above

sasikeda commented on resources/azure/security/network_application_security_group/network_application_security_group.tf:
```diff
@@ -0,0 +1,19 @@
+
+output "asg_id" {
+  value = azurerm_application_security_group.itself.id
+}
```
i think tools/setup_dev.sh is not run

sasikeda commented on resources/azure/security/network_application_security_group/scale_network_security_group copy.tfbackup:
```diff
@@ -0,0 +1,173 @@
+variable "security_group_name" {}
+variable "location" {}
+variable "resource_group_name" {}
+
```
This file should be deleted ?

praveenramuu committed:
Rebase: DNS update and filesystemupdate

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu commented:
Need to preform git rebase , Hence marked to draft

praveenramuu committed:
updated code with new data structs

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

sasikeda commented on azure_scale_templates/sub_modules/instance_template/locals.tf:
```diff
@@ -1,13 +1,332 @@
+          kms_key     = disk["kms_key"]
+          fs_name     = disk["fs_name"]
+          pool        = disk["pool"]
+          lun_no      = disk["lun_no"]
```
instead of passing from from flatterned. use the idx from here to lun_no, that will make it unique across filesystems pools.

praveenramuu replied:
Done

praveenramuu committed:
updated clun number

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

sasikeda commented on azure_scale_templates/sub_modules/bastion_template/outputs.tf:
```diff
@@ -3,8 +3,8 @@ output "bastion_instance_public_ip" {
-output "bastion_instance_id" {
-  value       = module.bastion_autoscaling_group[*].instance_ids
+output "bastion_instance_autoscaling_group_ref" {
+  value       = module.bastion_autoscaling_group[0].instance_ids[0]
```
keep the index inside the lower resource blocks or use a try block to avoid failure

sasikeda commented on azure_scale_templates/sub_modules/instance_template/locals.tf:
```diff
@@ -1,13 +1,332 @@
-  Local variables for the instance module.
-*/
 locals {
   cluster_type = (
```
the cluster type variable needs to be removed, as it is passed as variable instead of calculation

sasikeda commented on azure_scale_templates/sub_modules/instance_template/locals.tf:
```diff
@@ -1,13 +1,332 @@
+
+  tcp_port_scale_cluster         = ["22", "1191", "60000-61000", "47080", "4444", "4739", "9080", "9081", "80", "443"]
+  udp_port_scale_cluster         = ["47443", "4739"]
+  tcp_port_bastion_scale_cluster = ["22", "443"]
```
Is there any need for bastion ports to be defined here ?

praveenramuu replied:
This is used for allowing traffic from bastion to scale cluster

sasikeda replied:
In instance template, should be agnositic to jumphost and cloud vm scenarios. this rule should be removed.
I don't see this variable used as well (It must be flaged with setup_dev.sh)

praveenramuu replied:
This(tcp_port_bastion_scale_cluster) is already used in https://github.com/IBM/ibm-spectrum-scale-cloud-install/pull/474/files#diff-587018304ce4a2fce13789adb4f830040dc64945a2377a315c388ed63a659e8cR90

praveenramuu replied:
This is similar gcp rule as https://github.com/IBM/ibm-spectrum-scale-cloud-install/blob/next_gen/gcp_scale_templates/sub_modules/instance_template/main.tf#L31
```
# Create security rules to enable scale communication between bastion and scale instances
module "cluster_ingress_security_rule_using_jumphost_connection" {
```

sasikeda commented on azure_scale_templates/sub_modules/instance_template/locals.tf:
```diff
@@ -1,13 +1,332 @@
+  udp_port_scale_cluster         = ["47443", "4739"]
+  tcp_port_bastion_scale_cluster = ["22", "443"]
+
+  block_device_names = ["/dev/sdc", "/dev/sdd", "/dev/sdf", "/dev/sdg",
```
why device names are starting from sdc ? can they be started from sda in azure ?

praveenramuu replied:
Azure starts block device count from /dev/sdc only

sasikeda commented on azure_scale_templates/sub_modules/instance_template/locals.tf:
```diff
@@ -1,13 +1,332 @@
+    Ex: vm_list = ["vm-1", "vm-2", "vm-3",]
+*/
+resource "null_resource" "generate_compute_vm_name" {
+  count = local.compute_or_combined ? (var.total_compute_cluster_instances != null) ? var.total_compute_cluster_instances : 0 : 0
```
Use aws reference, these conditions are simplified

sasikeda commented on azure_scale_templates/sub_modules/instance_template/locals.tf:
```diff
@@ -1,13 +1,332 @@
+        type        = properties["type"]
+        iops        = properties["iops"]
+        throughput  = properties["throughput"]
+        lun_no      = properties["lun_no"]
```
remove lun_no from there

sasikeda commented on azure_scale_templates/sub_modules/instance_template/locals.tf:
```diff
@@ -1,13 +1,332 @@
+          "type"        = disk_details.block_device_volume_type
+          "iops"        = disk_details.block_device_iops
+          "throughput"  = disk_details.block_device_throughput
+          "lun_no"      = "${i + 1}"
```
remove lun_no from there

praveenramuu committed:
updated code as per review comment

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu commented:
Updated code as per review comment :
 - removed cluster_type variable
 - removed ' lun_no' where ever not required
 - removed bastion_scale_cluster_nsg
 - updated 'generate_compute_vm_name' null_resource w.r.t aws

praveenramuu committed:
update output bastion_instance_autoscaling_group_ref with try

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
removed proximity group for compute

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
removed unwanted files

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

sasikeda commented:
Few comments needs to be addressed.  Please have a check

praveenramuu committed:
udpated using_jumphost_connection to bastion rule

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>
